src/Npgsql/Internal/NpgsqlWriteBuffer.cs

```diff
@@ -494,7 +494,7 @@ public Stream GetStream()
         internal void WriteStringChunked(char[] chars, int charIndex, int charCount,
                                          bool flush, out int charsUsed, out bool completed)
         {
-            if (WriteSpaceLeft < _textEncoder.GetByteCount(chars, charIndex, 1, flush: false))
+            if (WriteSpaceLeft < _textEncoder.GetByteCount(chars, charIndex, Math.Min(2, charCount), flush: false))
```

Comment:
Is it slightly more correct to simply check if GetByteCount returns 0 (and if so, enter the condition and return)?

Reply:
Yeah, the above makes no sense - it would just send us into an infinite loop, since count would always be 1 and GetByteCount would always return 0. Merging as-is.